### PR TITLE
refactor: `Chunk.modules` => `Chunk.mapModules`  (`webpack v3.0.0`)

### LINKED ISSUE #529: feat: DeprecationWarning: Chunk.modules in Webpack 3.x
Webpack 3.0-rc was released and this plugin currently doesn't support it.

I haven't tested it, it might be just a question of updating the peerDependency.

### PR DESCRIPTION
**Pull request by MirrorBytes**
Refactors to use webpack 3.x `Chunk.mapModules` 

Closes #529 

> Squashed commit message body for merge
```

- refactor: DeprecationWarning: Chunk.modules

BREAKING CHANGE: Updates to `Chunk.mapModules`. This release is not backwards compatible with `Webpack 2.x` due to breaking changes in webpack/webpack#4764
```

### CODE REVIEW

**Comment by MirrorBytes** [comment]
Should resolve #529 if merged.

File: index.js [hunk 1]
```diff
@@ -269,7 +269,7 @@ ExtractTextPlugin.prototype.apply = function(compiler) {
 			async.forEach(chunks, function(chunk, callback) {
 				var extractedChunk = extractedChunks[chunks.indexOf(chunk)];
 				var shouldExtract = !!(options.allChunks || isInitialOrHasNoParents(chunk));
-				async.forEach(chunk.modules.slice(), function(module, callback) {
+				async.forEach(chunk.mapModules(function(c) { return c; }), function(module, callback) {
```
Comment (d3viant0ne) [inline index.js]: Having just refactored this entire plugin, there are 4/5 instances of `chunk.modules.something`

**Comment by michael-ciniawsky** [comment]
This will land in the `v2.x` fixing the `DeprecationWarning` , before `webpack-defaults` (`v3.0.0`) lands? :)  I'm a bit confused 😛

**Comment by d3viant0ne** [comment]
@michael-ciniawsky - This change is not backwards compatible with 2.x hence why it's targeting the `features/webpack3`.

This can merge into that target branch when ready & I can rebase in the changes to the defaults PR.

**Comment by MirrorBytes** [comment]
@d3viant0ne I'll change all of the instances accordingly. Sound good?

**Comment by michael-ciniawsky** [comment]
@MirrorBytes What is missing here, did you find the remaining places, where this needs to be updated as @d3viant0ne suggested?

**Comment by evilebottnawi** [comment]
@michael-ciniawsky maybe we can check webpack version and use difference behavior without breaking changes?

**Comment by michael-ciniawsky** [comment]
@evilebottnawi How would this look like ?  There is `this.version` (at least in the Loader API), so maybe

```js
function mapModules (chunk) {
   if (this.version >= 3) return chunk.mapModule((c) => c)
   return chunk.modules.slice()
}
```

or check the `chunk`  for extisting methods  would work 😛

**Comment by evilebottnawi** [comment]
@michael-ciniawsky seems good, but need wait what says @d3viant0ne

**Comment by michael-ciniawsky** [comment]
Fixes #494

**Comment by d3viant0ne** [comment]
@MirrorBytes - This will need to be rebased now that the defaults PR has landed.

Don't know how good you are with git, if you aren't comfortable doing it just give me write permission ( hit edit in this pr & you should see a check box ) and I will rebase it for you.

File: index.js (hunk 1 above)
Comment (JLHwung) [inline index.js]: How about `function (m) { return m; }` for precise initials?

**Comment by themre** [comment]
I hear appveyor has some issues lately. perhaps someone else can run it on windows to release it.

**Comment by bobvanderlinden** [comment]
I rebased this PR on `feature/webpack3` and changed the remaining references to `Chunk.modules` to `Chunk.mapModules`. The resulting version can be found here: https://github.com/bobvanderlinden/extract-text-webpack-plugin/commits/pr-543

With the additional changes, the test does not output any warnings like it did before:

```
(node:3880) DeprecationWarning: Chunk.modules is deprecated. Use Chunk.getNumberOfModules/mapModules/forEachModule/containsModule instead.
```

I ran the tests on Windows 10, node v6.4.0, webpack 3.0.0.
1 test failed:

```
Expected value to equal:
  "body {
        correct: a;
}
body {
        correct: b;
}

/*# sourceMappingURL=file.css.map*/"
Received:
  "body {
        correct: a;
}
body {
        correct: b;
}

/*# sourceMappingURL=file.css.map*/"

Difference:

- Expected
+ Received

@@ -2,7 +2,7 @@
        correct: a;
 }
 body {
        correct: b;
 }
-
+
 /*# sourceMappingURL=file.css.map*/
```

Note though that this seems unrelated to this PR as the test also fails on `feature/webpack3`. I'm guessing it's an `\r\n` vs `\n` problem?

**Comment by d3viant0ne** [comment]
I'll try to take a look at the windows issue tonight.

File: index.js (hunk 1 above)
Comment (d3viant0ne) [inline index.js]: `c === chunk` the naming convention is as it should be

Comment (JLHwung) [inline index.js]: `mapModules` map functions on each modules so the iteratee will be called with each module. See https://github.com/webpack/webpack/pull/4764/files#diff-8941681d920ac4feda44f522b986c2d0R115

**Comment by c0b41** [comment]
any status?

**Comment by cklmercer** [comment]
Not sure if this is related but I recently started encountering the following issue on Windows 10.

```
TypeError: chunk.mapModules is not a function
    at C:\Users\me\Code\xxx\xxx.com\node_modules\webpack\lib\HotModuleReplacementPlugin.js:59:46
    at Array.forEach (native)
    at Compilation.<anonymous> (C:\Users\me\Code\xxx\xxx.com\node_modules\webpack\lib\HotModuleReplacementPlugin.js:58:16)
    at Compilation.applyPlugins2 (C:\Users\me\Code\xxx\xxx.com\node_modules\tapable\lib\Tapable.js:82:14)
    at sealPart2 (C:\Users\me\Code\xxx\xxx.com\node_modules\webpack-dev-server\node_modules\webpack\lib\Compilation.js:629:10)
    at next (C:\Users\me\Code\xxx\xxx.com\node_modules\tapable\lib\Tapable.js:138:11)
    at Compilation.compilation.plugin (C:\Users\me\Code\xxx\xxx.com\node_modules\webpack-dev-server\node_modules\webpack\lib\ProgressPlugin.js:109:6)
    at Compilation.applyPluginsAsyncSeries (C:\Users\me\Code\xxx\xxx.com\node_modules\tapable\lib\Tapable.js:142:13)
    at Compilation.seal (C:\Users\me\Code\xxx\xxx.com\node_modules\webpack-dev-server\node_modules\webpack\lib\Compilation.js:579:8)
    at C:\Users\me\Code\xxx\xxx.com\node_modules\webpack-dev-server\node_modules\webpack\lib\Compiler.js:493:16
```

**Comment by d3viant0ne** [comment]
This has been taken care of in another PR given the time pressure to get etwp stable in Webpack 3.x

**Pull request closed without merging**